sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -299,20 +251,18 @@ private[jdbc] class JDBCRDD(
     // fully-qualified table name in the SELECT statement.  I don't know how to
     // talk about a table in a completely portable way.
 
-    val myWhereClause = getWhereClause(part)
-
-    val myTableSampleClause: String = if (sample.nonEmpty) {
-      JdbcDialects.get(url).getTableSample(sample.get)
-    } else {
-      ""
-    }
+    val builder = dialect
+      .getJdbcSQLQueryBuilder(options)
+      .withColumns(columns)
+      .withPredicates(predicates, part)
+      .withSortOrders(sortOrders)
+      .withLimit(limit)
+      .withOffset(offset)
 
-    val myLimitClause: String = dialect.getLimitClause(limit)
-    val myOffsetClause: String = dialect.getOffsetClause(offset)
+    groupByColumns.foreach(builder.withGroupByColumns)
```

Comment:
`withXXX` returns a new builder. We can's assume the implementation always returns `this`

let's do
```
var builder = ...
groupByColumns.foreach { groupBy =>
  builder = ...
}
```

Reply:
Yeah. I make a mistake.